emission/tests/modellingTests/modellingTestAssets.py

```diff
@@ -1,35 +1,72 @@
-
+import emission.core.common as ecc
 import emission.core.wrapper.entry as ecwe
 import time 
 import math
 
+def generate_random_point():
+    """Generate a completetly random point valid WGS84 latitiude and longtidude.   
+    CAUTION : In order to save trips, GeoJSON requires points in [lon,lat] format"""    
+    lat=random.uniform(-90,90)
+    lon=random.uniform(-180,180)
+    return [lon,lat]
+
+def generate_nearby_random_points(ref_coords,threshold):
+    """
+    Generate valid WGS84 latitiude and longtidude in threshold(m) proximity to
+    ref coordinates.
+    """
+    #convert given threshold in m to approx WGS84 coord dist.
+    thresholdInWGS84 = threshold* (0.000001/0.11)
+    
+    #generate a random coordinate in threshold's limit around the ref points. 
+    dx=random.uniform(-thresholdInWGS84/2,thresholdInWGS84/2)
+    dy=random.uniform(-thresholdInWGS84/2,thresholdInWGS84/2)
+
+    #This basically gives a way to sample a point from within a square of length thresholdInWGS84 
+    # around the ref. point.  
+    return [ref_coords[0] +dx , ref_coords[1] +dy]
 
 def generate_trip_coordinates(
-    ref_coords: Tuple[float, float], 
-    within_threshold: bool,
-    threshold: float, 
-    max: float = 0.1  # approx. 10km in WGS84 
+    points_list: list[float],
+    ref_coords, 
+    InsideThreshold: bool,
+    threshold_meters: float, 
     ) -> Tuple[float, float]:
-    """generates trip coordinate data to use when mocking a set of trip data.
+    """generates trip coordinate data to use when mocking a set of trip data.i
+    If the coordinate generated  is to be binned together, it is generated in proximity of
+    the previous points in the points_list. Otherwise, if this point is not to be binned together,
+    keep generating a random trip unless we find one that would not bin with previously
+    accepeted trips.
 
-    :param ref_coords: reference coordinates to use as the center of the sampling circle
-    :param within_threshold: how many of these trips are within some distance threshold
-    :param threshold: the distance threshold, in WGS84
-    :param max: max distance, in WGS84, defaults to 0.1 (approx. 10km)
+    :param points_list: list of all the previoushlt selected points 
+    :param within_threshold: is this point  supposed to be within some distance threshold
+    :param threshold_meters: the distance threshold, in meters
     :return: generated coordinate pairs sampled in a 
              circle from some coordinates up to some threshold
     """
-    angle = 2 * math.pi * random.random()
-    radius_threshold = threshold / 2
-    radius = random.uniform(0, radius_threshold) if within_threshold else random.uniform(radius_threshold, max)
-    x = radius * math.cos(angle) + ref_coords[0]
-    y = radius * math.sin(angle) + ref_coords[1]
-    return (x, y)
+    # if the point is to be generated within a threshold and it's not the first point
+    if InsideThreshold and points_list:
```

Comment:
nit: by most naming conventions, variables in python start with lowercase, classes start with upper case